src/utilities/BatchTableHierarchyExtension.js

```diff
@@ -0,0 +1,122 @@
+import { parseBinArray } from './FeatureTable.js';
+
+export class BatchTableHierarchyExtension {
+
+	constructor( batchTable ) {
+
+		this.batchTable = batchTable;
+
+		const extensionHeader = batchTable.header.extensions[ '3DTILES_batch_table_hierarchy' ];
+
+		this.classes = extensionHeader.classes;
+		for ( const classDef of this.classes ) {
+
+			const instances = classDef.instances;
+			for ( const property in instances ) {
+
+				classDef.instances[ property ] = this._parseProperty( instances[ property ], classDef.length );
+
+			}
+
+		}
+
+		this.instancesLength = extensionHeader.instancesLength;
+
+		this.classIds = this._parseProperty( extensionHeader.classIds, this.instancesLength );
+
+		if ( extensionHeader.parentCounts ) {
+
+			this.parentCounts = this._parseProperty( extensionHeader.parentCounts, this.instancesLength );
+
+		} else {
+
+			this.parentCounts = new Array( this.instancesLength ).fill( 1 );
```

Comment:
Why default this to 1? Shouldn't it be defaulted to 0 if there are no parents specified?

Reply:
I agree it feels more logical but the specification defaults `parentCounts` to an array of ones ([here](https://github.com/CesiumGS/3d-tiles/blob/87218a7c2030890b5200cf16cb466dbbd1480a20/extensions/3DTILES_batch_table_hierarchy/README.md?plain=1#L190))